final class org.apache.poi.ss.formula.functions.NumericFunction$27 extends org.apache.poi.ss.formula.functions.NumericFunction$TwoArg
{

    void <init>()
    {
        org.apache.poi.ss.formula.functions.NumericFunction$27 r0;

        r0 := @this: org.apache.poi.ss.formula.functions.NumericFunction$27;

        specialinvoke r0.<org.apache.poi.ss.formula.functions.NumericFunction$TwoArg: void <init>()>();

        return;
    }

    protected double evaluate(double, double) throws org.apache.poi.ss.formula.eval.EvaluationException
    {
        org.apache.poi.ss.formula.eval.ErrorEval $r1;
        org.apache.poi.ss.formula.functions.NumericFunction$27 r2;
        byte $b0, $b1;
        org.apache.poi.ss.formula.eval.EvaluationException $r0;
        double d0, d1, $d2;

        r2 := @this: org.apache.poi.ss.formula.functions.NumericFunction$27;

        d1 := @parameter0: double;

        d0 := @parameter1: double;

        $b0 = d0 cmpl 0.0;

        if $b0 != 0 goto label2;

        $b1 = d1 cmpl 0.0;

        if $b1 != 0 goto label1;

        return 0.0;

     label1:
        $r0 = new org.apache.poi.ss.formula.eval.EvaluationException;

        $r1 = <org.apache.poi.ss.formula.eval.ErrorEval: org.apache.poi.ss.formula.eval.ErrorEval DIV_ZERO>;

        specialinvoke $r0.<org.apache.poi.ss.formula.eval.EvaluationException: void <init>(org.apache.poi.ss.formula.eval.ErrorEval)>($r1);

        throw $r0;

     label2:
        $d2 = staticinvoke <org.apache.poi.ss.formula.functions.MathX: double floor(double,double)>(d1, d0);

        return $d2;
    }
}
